### ISSUE TITLE
Can't parse svg, if <svg> start tag appears after 300 bytes

### ISSUE THREAD

**Issue by thomas15k** [lovell/sharp#1252]
**previous title: "XML parser doesn't understand custom entities" was due to wrong conclusion.**

This header gives Error: Input buffer contains unsupported image format

```
<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 17.0.1, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd" [
        <!ENTITY ns_extend "http://ns.adobe.com/Extensibility/1.0/">
        <!ENTITY ns_ai "http://ns.adobe.com/AdobeIllustrator/10.0/">
        <!ENTITY ns_graphs "http://ns.adobe.com/Graphs/1.0/">
        <!ENTITY ns_vars "http://ns.adobe.com/Variables/1.0/">
        <!ENTITY ns_imrep "http://ns.adobe.com/ImageReplacement/1.0/">
        <!ENTITY ns_sfw "http://ns.adobe.com/SaveForWeb/1.0/">
        <!ENTITY ns_custom "http://ns.adobe.com/GenericCustomNamespace/1.0/">
        <!ENTITY ns_adobe_xpath "http://ns.adobe.com/XPath/1.0/">
]>
<svg version="1.1" id="Layer_1" xmlns:x="&ns_extend;" xmlns:i="&ns_ai;" xmlns:graph="&ns_graphs;"
         xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="262.681px"
         height="262.681px" viewBox="0 0 262.681 262.681" enable-background="new 0 0 262.681 262.681" xml:space="preserve">
```

**Comment by thomas15k** [comment]
[with_entities_fails.svg.txt](https://github.com/lovell/sharp/files/2072939/with_entities_fails.svg.txt)
[without_entities_works.svg.txt](https://github.com/lovell/sharp/files/2072940/without_entities_works.svg.txt)

**Comment by lovell** [comment]
Hello, this is related to https://github.com/jcupitt/libvips/pull/543 - the `<svg` bit has to appear within the first 300 bytes.

**Comment by lovell** [comment]
sharp v0.21.0 is now available with a prebuilt libvips v8.7.0 that contains the upstream fix.
